reactor-netty-core/src/test/java/reactor/netty/transport/ClientTransportTest.java

```diff
@@ -132,6 +133,80 @@ void testClientTransportWarmup() {
 		}
 	}
 
+	@Test
+	void testCreateClientWithHttpProxy() {
+		Properties properties = new Properties();
+		properties.setProperty(ProxyProvider.HTTP_PROXY_HOST, "host");
+
+		TestClientTransport transport = createTestTransportForProxy()
+				.proxyWithSystemProperties(properties);
+
+		assertThat(transport.configuration().proxyProvider()).isNotNull();
+		assertThat(transport.configuration().proxyProvider().getType()).isEqualTo(ProxyProvider.Proxy.HTTP);
+	}
+
+	@Test
+	void testCreateClientWithHttpsProxy() {
+		Properties properties = new Properties();
+		properties.setProperty(ProxyProvider.HTTPS_PROXY_HOST, "host");
+
+		TestClientTransport transport = createTestTransportForProxy()
+				.proxyWithSystemProperties(properties);
+
+		assertThat(transport.configuration().proxyProvider()).isNotNull();
+		assertThat(transport.configuration().proxyProvider().getType()).isEqualTo(ProxyProvider.Proxy.HTTP);
+	}
+
+	@Test
+	void testCreateClientWithSock5Proxy() {
```

Comment:
there should probably be a separate test for default version (this test) vs explicit version 5, like there is one for explicit version 4

Reply:
added